applications/FluidDynamicsApplication/custom_elements/data_containers/two_fluid_fractional_navier_stokes/two_fluid_navier_stokes_fractional_data.h

```diff
@@ -0,0 +1,381 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Uxue Chasco
+//
+
+#pragma once
+
+#define KRATOS_TWO_FLUID_NAVIER_STOKES_FRACTIONAL_DATA_H
+
+#include "includes/constitutive_law.h"
+
+#include "fluid_dynamics_application_variables.h"
+#include "custom_elements/data_containers/fluid_element_data.h"
+#include "utilities/element_size_calculator.h"
+#include "custom_utilities/fluid_element_utilities.h"
+
+namespace Kratos {
+
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos classes
+///@{
+
+template< size_t TDim, size_t TNumNodes >
+class TwoFluidNavierStokesFractionalData : public FluidElementData<TDim,TNumNodes, true>
+{
+public:
+
+///@name Type Definitions
+///@{
+
+using NodalScalarData = typename FluidElementData<TDim,TNumNodes, true>::NodalScalarData;
+using NodalVectorData = typename FluidElementData<TDim,TNumNodes, true>::NodalVectorData;
+using ShapeFunctionsType = typename FluidElementData<TDim, TNumNodes, true>::ShapeFunctionsType;
+using ShapeDerivativesType = typename FluidElementData<TDim, TNumNodes, true>::ShapeDerivativesType;
+using MatrixRowType = typename FluidElementData<TDim, TNumNodes, true>::MatrixRowType;
+typedef Geometry<Node> GeometryType;
+typedef GeometryType::ShapeFunctionsGradientsType ShapeFunctionsGradientsType;
+
+static constexpr std::size_t BlockSize = TDim + 1;
```

Comment:
Shouldn't this be `TDim`. Note that this is a potential source of memory errors...

Reply:
I would say that in this case for the second problem of N.S the block size is Tdim + 1, equal, to the original case of two fluid NS bdf2 and alpha.